src/components/cluster-deployments/ClusterDeploymentWizard.tsx

```diff
@@ -0,0 +1,203 @@
+import { AgentClusterInstallKind, ClusterDeploymentKind, ClusterImageSetKind } from '../../kind';
+import { AgentClusterInstallK8sResource, ClusterDeploymentK8sResource } from '../types';
+import { getAICluster } from '../ai-utils';
+import { appendPatch, getClusterDeployment, getPullSecretResource } from '../../k8s';
+import { getAgentClusterInstall } from '../../k8s/agentClusterInstall';
+
+type ClusterDeploymentWizardProps = {
+  history: RouteComponentProps['history'];
+  namespace?: string;
+  clusterDeployment?: ClusterDeploymentK8sResource;
+  agentClusterInstall?: AgentClusterInstallK8sResource;
+};
+
+const ClusterDeploymentWizard: React.FC<ClusterDeploymentWizardProps> = ({
+  history,
+  namespace,
+  clusterDeployment: queriedClusterDeployment,
+  agentClusterInstall: queriedAgentClusterInstall,
+}) => {
+  const [clusterDeploymentModel] = useK8sModel(ClusterDeploymentKind);
+  const [agentClusterInstallModel] = useK8sModel(AgentClusterInstallKind);
+  const [secretModel] = useK8sModel('core~v1~Secret');
+
+  const [clusterDeployment, setClusterDeployment] =
+    React.useState<ClusterDeploymentK8sResource>(queriedClusterDeployment);
+  const [agentClusterInstall, setAgentClusterInstall] =
+    React.useState<AgentClusterInstallK8sResource>(queriedAgentClusterInstall);
+
+  React.useEffect(() => {
+    setClusterDeployment(queriedClusterDeployment);
+  }, [clusterDeployment !== queriedClusterDeployment]);
+  React.useEffect(() => {
+    setAgentClusterInstall(queriedAgentClusterInstall);
+  }, [agentClusterInstall !== queriedAgentClusterInstall]);
+
+  const defaultPullSecret = ''; // Can be retrieved from c.rh.c . We can not query that here.
+
+  const [clusterImageSets] = useK8sWatchResource<K8sResourceCommon[]>({
+    kind: ClusterImageSetKind,
+    namespaced: false,
+    isList: true,
+  });
+  const ocpVersions = (clusterImageSets || []).map(
+    (clusterImageSet, index): Types.OpenshiftVersionOptionType => {
+      return {
+        label: clusterImageSet.metadata.name,
+        value: clusterImageSet.metadata.name, // TODO(mlibra): probably wrong but what is expected here?
+        default: index === 0,
+        supportLevel: 'beta', // TODO(mlibra): Map from label "channel"
+      };
+    },
+  );
+
+  const [clusterDeployments] = useK8sWatchResource<ClusterDeploymentK8sResource[]>({
+    kind: ClusterDeploymentKind,
+    namespace, // TODO(mlibra): Double check that we want to validate cluster name for namespace-only (and not cluster-scope, mind prvileges)
+    namespaced: true,
+    isList: true,
+  });
+  const usedClusterNames = (clusterDeployments || [])
+    .filter((cd) => !clusterDeployment || clusterDeployment.metadata.uid !== cd.metadata.uid)
+    .map((cd): string => `${cd.metadata.name}.${cd.spec?.baseDomain}`);
+
+  const onClusterCreate = React.useMemo(
+    () =>
+      async ({ pullSecret, openshiftVersion, ...params }: ClusterDeploymentDetailsValues) => {
```

Comment:
```suggestion
  const onClusterCreate = React.useCallback(async ({ pullSecret, openshiftVersion, ...params }: ClusterDeploymentDetailsValues) => {
```

Reply:
same for other `useMemo`-s below. Not a big deal at this point though

Reply:
done